### ISSUE TITLE
Proposal: --stdin-filename gives .eslintrc files to load

### ISSUE THREAD

**Issue by fflorent** [eslint/eslint#3628]
I made a git hook in which I check the indexed content of the code before committing it:
````bash
    git diff --cached "$file" | eslint --stdin --stdin-filename "$file"
````

However, I also have `.eslintrc` files in my project that isn't loaded when launching the linter.

As we have the `--no-eslintrc` option, could `--stdin-filename` hint what `.eslintrc` files to load to eslint?

Edit: using eslint v0.20.0

Thanks in advance,
Florent

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by fflorent** [comment]
As you seem to be much busy, I can try to implement the feature by my own if you're OK with that.

Florent

**Comment by platinumazure** [comment]
Does `--stdin-filename` load `.eslintrc` files with the latest ESLint?

**Comment by ilyavolodin** [comment]
@fflorent If I'm understanding your question correctly, I don't think it's possible to tie-in stdin-filename to a .eslintrc file. stdin-filename is completely arbitrary, It can be anything. I guess you could check if there's a .eslintrc file in the current directory where eslint is being executed from, but that seems pretty random decision. Why don't you just use the following command:
```sh
git diff --cached "$file" | eslint --stdin --stdin-filename "$file" -c .eslintrc
```
This way you can always supply correct .eslintrc file.

**Comment by nzakas** [comment]
Just to make sure I'm understanding this correctly, what you're saying is that when you use `--stdin-filename`, your `.eslintrc` file isn't being loaded?

Can you try with the latest version of ESLint? From my local testing, we do use `--stdin-filename` as the key for loading the corresponding `.eslintrc` files.

Of course, if I'm misunderstanding, please let me know what it is you're reporting.

**Comment by fflorent** [comment]
> Can you try with the latest version of ESLint? From my local testing, we do use --stdin-filename as the key for loading the corresponding .eslintrc files.

Works like a charm. Thanks !

Florent
